### PR TITLE
Support async iterables

### PR DESCRIPTION
**Pull request by papb**
This PR adds support for converting async iterables into readable streams.

Note: build is failing for Node 8, so I guess you'll want to make a major version bump first dropping support for it?

### CODE REVIEW

File: index.test-d.ts [hunk 1]
```diff
@@ -22,8 +34,12 @@ intoStream(Promise.resolve(new Uint8Array(Buffer.from('unicorn').buffer))).pipe(
 	process.stdout
 );
 
+intoStream(asyncGeneratorFrom(unicornArray)()).pipe(process.stdout);
+intoStream(asyncIterableFrom(unicornArray)).pipe(process.stdout);
+intoStream(Promise.resolve(asyncIterableFrom(unicornArray))).pipe(process.stdout);
+
 const object = {foo: true};
-const objectArray = new Set([object, {bar: true}]);
+const objectArray = [object, {bar: true}];
```
Comment (papb) [inline index.test-d.ts:42]: Should I open a separate PR for this fix?

File: readme.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # into-stream [![Build Status](https://travis-ci.org/sindresorhus/into-stream.svg?branch=master)](https://travis-ci.org/sindresorhus/into-stream)
 
-> Convert a string/promise/array/iterable/buffer/typedarray/arraybuffer/object into a stream
+> Convert a string/promise/array/iterable/asynciterable/buffer/typedarray/arraybuffer/object into a stream
```
Comment (papb) [inline readme.md:3]: I was unsure of which casing to use here...

File: index.test-d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.test-d.ts:42]: Nah, it's fine.

File: readme.md (hunk 2 above)
Comment (sindresorhus) [reply readme.md:3]: This is fine

**Comment by sindresorhus** [comment]
Nice work 👍🏻

**Pull request merged**